docs/build/node.md

```diff
@@ -0,0 +1,71 @@
+# Node use in the Keyman project
+
+For a given release cycle, we select and pin to a specific version of node.js,
+and try to use that throughout. The version in use can be found in
+`package.json/engines.node`.
+
+If we encounter a blocking issue, we will upgrade to a known-good release, but
+this is exceptional.
+
+Generally, as a developer, if you use the same major version as is found in
+package.json, you probably won't have significant issues. However, if you are
+switching between stable branches and alpha branches, you may find the required
+node version changes.
+
+## Automatic node version selection
+
+The Keyman build system can be configured to manage the node.js version for you.
+This does have several caveats, so it is not enabled by default.
+
+When this is enabled, the build system will download the required node version
+and make it available on the PATH, and it will also stop the build if the node
+version mismatches. This happens in `build.sh configure` steps for any
+Typescript/Javascript project.
+
+To enable automatic node version selection, add the variable `KEYMAN_USE_NVM=1`
+to your environment.
+
+### Caveats
+
+nvm (for macOS/Linux) and nvm-windows use somewhat different paradigms, so the
+caveats vary per platform.
+
+### Caveats on macOS/Linux
+
+On macOS/Linux, nvm is provided as a shell function that modifies the current
+environment PATH to make a specific node version available -- and so this does
+not affect unrelated processes.
+
+1. You must use [nvm](https://github.com/nvm-sh/nvm) (macOS/Linux) to install
+   and manage Node versions.
+2. The Keyman build environment will create a symlink at `~/.keyman/node`
+   pointing to the version of node selected by nvm.
+3. You should add `$HOME/.keyman/node` to the front of your `PATH` variable,
+   e.g. in `~/.bash_profile`. This means that the node version _will_ be set for
+   the entire system, unlike with standard nvm usage. This allows build scripts
+   to run without calling `nvm` for each invocation.
```

Comment:
This feels like a bit of a big hammer, though I suppose this _is_ why it's being kept locked behind `KEYMAN_USE_NVM=1`.

If only we could easily rewrite the script's working `PATH` after we've already launched it so that we could set this in place for just our own scripts.

Reply:
If you can see a better way then I'd be happy to try that -- but this hammer seems to work well and is hopefully better than many of the other options. It _does_ persist between builds, which is both good and bad (is a leaving an specific version as default that we have to manually update any better?)

My primary concern is that we may want to update some of the build configs that don't use nvm (e.g. keyboards and models builds) so that they can also use a specific version of node. It's not all that important because they'll likely continue to work anyway, but it would be tidier!